src/main/java/edu/bu/metcs673/trackr/user/TrackrUserController.java

```diff
@@ -15,106 +13,94 @@
-    public ResponseEntity<GenericApiResponse> login(
-            @Valid @RequestBody TokenRetrievalDTO userLogin,
-            HttpServletResponse response) {
-
-        try {
-            Authentication auth = authenticationManager.authenticate(
-                    new UsernamePasswordAuthenticationToken(
-                            userLogin.getUsername(),
-                            userLogin.getPassword()
-                    )
-            );
-            String token = jwtUtil.generateToken(((UserDetails) auth.getPrincipal()).getUsername());
-            JSONObject tokenObj = createTokenObject(token);
-
-            // Return the JWT to the browser as cookie
-            Cookie cookie = new Cookie(JWT_COOKIE_NAME, token);
-
-            // We set HttpOnly so that JavaScript cannot read it top prevent XSS attack
-            cookie.setHttpOnly(true);
-
-            // Set path to root so that it's available to the whole application domain
-            cookie.setPath(JWT_COOKIE_PATH);
-
-            // Set max age to be 15 minutes
-            cookie.setMaxAge(JWT_COOKIE_MAX_AGE_MINUTES);
-
-            // Add Servlet Response to return to browser
-            response.addCookie(cookie);
-
-            return ResponseEntity.ok()
-                    .body(GenericApiResponse.successResponse(CommonConstants.NEW_JWT_TOKEN, tokenObj));
-        } catch (AuthenticationException e) {
-            return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
-                    .body(GenericApiResponse.errorResponse(CommonConstants.INVALID_CREDENTIALS));
-        }
-    }
-
-    /**
-     * Helper method to present the JWT Token value to the user in a more organized way.
-     *
-     * @param token JSON Web Token string
-     * @return A JSON object
-     */
-    static JSONObject createTokenObject(String token) {
-        JSONObject tokenObj = new JSONObject();
-        tokenObj.put("jwtToken", token);
-        return tokenObj;
-    }
+	@Autowired
+	private TrackrUserService userService;
+
+	@Autowired
+	private JWTUtil jwtUtil;
+
+	@Autowired
+	private AuthenticationManager authenticationManager;
+
+	/**
+	 * Using provided request body, creates a new record in the USERS table. Request
+	 * body is validated, and an encrypted password value is saved. Returns a
+	 * success message with a JWT token.
+	 * 
+	 * @param userInput
+	 * @return GenericApiResponse
+	 */
+	@PostMapping
+	public ResponseEntity<GenericApiResponse> createUser(@Valid @RequestBody TrackrUserDTO userInput) {
+		String token = userService.createUser(userInput);
+
+		JSONObject tokenObj = createTokenObject(token);
+
+		return new ResponseEntity<GenericApiResponse>(
+				GenericApiResponse.successResponse(CommonConstants.CREATE_USER_SUCCESS, tokenObj), HttpStatus.OK);
+	}
+
+	/**
+	 * Using the provided valid request body, authenicate the user and if
+	 * credentials are correct, then provide a new JWT token for other API use.
+	 * 
+	 * @param userLogin
+	 * @return
+	 */
```

Comment:
This looks like the old file and it's going to break the UI. Can you please update with the file from development branch?

Reply:
pulled most recent code, and updated the PR